bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/SkijaGC.java

```diff
@@ -465,11 +464,25 @@ public void drawRoundRectangle(int x, int y, int width, int height, int arcWidth
 	}
 
 	public void drawString(String string, int x, int y) {
-		System.err.println("WARN: Not implemented yet: " + new Throwable().getStackTrace()[0]);
+		int topLeftTextBoxYPosition = DPIUtil.autoScaleUp(y);
+		float heightOfTextBoxConsideredByClients = font.getMetrics().getHeight();
+		float heightOfSymbolToCenter = baseSymbolHeight;
+		performDrawText(paint -> {
+			surface.getCanvas()
+					.drawString(
+							string, (int) DPIUtil.autoScaleUp(x), (int) (topLeftTextBoxYPosition
+									+ heightOfTextBoxConsideredByClients / 2 + heightOfSymbolToCenter / 2),
+							font, paint);
+		});
```

Comment:
Can we factor out the common code between `drawString()` and `drawText()`. Then in case we find something not properly working with the current code, fixes will affect both methods without knowing that there is a duplication of the logic.